scripts/wordlist_to_vocab.py

```diff
@@ -37,7 +37,7 @@
 args = parser.parse_args()
 
 # read in the weights.
-words = open(args.wordlist, "r").readlines()
+words = open(args.wordlist, "r", encoding="utf-8").readlines()
```

Comment:
Sorry for the delay. I was busy with some coursework and exams...
I have found a bug introduced by this line of modification.

The bug appears later at line 97: 
```
print(output[i].encode('utf-8'), i)
```
When the printed output is piped to a file, it may cause some error like this:
```
UnicodeEncodeError: 'ascii' codec can't encode character '\u0117' in position 9: ordinal not in range(128)
```

This is because (I guess), after the modification -- changing to Python 3 and adding `encoding="utf-8"` here -- the string is now represented internally as a sequence of bytes that [corresponds to its Unicode](https://stackoverflow.com/questions/4987327/how-do-i-check-if-a-string-is-unicode-or-ascii). 
Then, when printing and when the encoding of the output is not defined, [it defaults to 7-bit ASCII](http://www.macfreek.nl/memory/Encoding_of_Python_stdout). Thus cause the error.

A way to solve this might be, at line 97 (_and all print statements_), simply change it to:
```
print(output[i].encode('utf-8'), i)
```
as examples [here](https://stackoverflow.com/questions/3597480/how-to-make-python-3-print-utf8). However, this causes new problem for line 56! The constant strings such as '\<s\>' will now be printed as 'b'\<s\>'', as the same problem [here](https://stackoverflow.com/questions/25127673/how-to-print-utf-8-to-console-with-python-3-4-windows-8) or [here](https://stackoverflow.com/questions/6269765/what-does-the-b-character-do-in-front-of-a-string-literal).

Thus, for this pull request, perhaps we still need some mechanism that can output correctly and uniformly for different strings. Perhaps we can just write bytes to stdout, as suggested [here](https://stackoverflow.com/a/3603160):
```
import sys
sys.stdout.buffer.write(TestText2)
```
But I guess you and @danpovey may have come across similar situations before and have a solution. I might have complicated the problem.

I have not check why the previous version of codes in Python 2.7 does not have the same problem, but Python [2.7](https://docs.python.org/2/library/functions.html#open) and [3](https://docs.python.org/3/library/functions.html#open) seems to have very different io and string handling mechanism.

Another helpful reference is [here](https://docs.python.org/3/howto/unicode.html).
If I am not clear, please let me know.

Reply:
Thanks for finding the problem.
This kind of thing is locale-dependent, so it's good that you tested it.  
I think the right way to solve this is to reopen the stdout stream / set it to utf-8 mode.  There is some kind of special invocation that can do this.

Reply:
Thanks for the suggestions.
Yes, this problem seems to happen only when `wordlist_to_vocab.py` is invoked by Kaldi's scripts on the clsp grid. We can check in this situation:
```
sys.stdout.encoding = ANSI_X3.4-1968
```
Otherwise, if I run `wordlist_to_vocab.py` _**alone**_ from the console of the grid, or on my laptop, it runs correctly, with:
```
sys.stdout.encoding = UTF-8
```
Thus, I guess Kaldi must have setup something before hand, so the Python3's behavior is affected.

To set `sys.stdout.encoding`, we might not set the environment variables, like:
```
export PYTHONIOENCODING=utf-8
```
as suggested [here](https://docs.python.org/3/library/sys.html#sys.stdout) or [here](https://stackoverflow.com/a/6362647), because this might cause other problems in the downstream. Instead, we hope to solve this within Python in runtime.
Thus, a more suitable way can be:
```
import codecs
sys.stdout = codecs.getwriter("utf-8")(sys.stdout.detach())
```
as suggested [here](https://stackoverflow.com/a/4374457) or [here](https://stackoverflow.com/a/1169209). I've test this solution and it runs correctly for the moment. Is this what you want? @danpovey

Another way is like:
```
sys.stdout.reconfigure(encoding='utf-8')
```
as suggested [here](https://stackoverflow.com/a/52372390), but it seems this is new since Python 3.7 and may not be compatible for earlier versions.